src/colorize.cr

```diff
@@ -1,3 +1,91 @@
+# With Colorize you can change the fore- and background colors and text decorations when rendering text
+# on terminals supporting ANSI escape codes. It adds the `colorize` method to `Object` and thus all classes
+# as its main interface, which calls `to_s` and surrounds it with the necessary escape codes
+# when it comes to obtaining a string representation of the object. 
+#
+# Its first argument changes the foreground color:
+# ```crystal
+# require "colorize"
+#
+# "foo".colorize(:green)
+# 100.colorize(:red)
+# [1,2,3].colorize(:blue)
+# ```
+#
+# There are alternative ways to change the foreground color:
+# ```crystal
+# "foo".colorize.fore(:green)
+# "foo".colorize.green
+# ```
+#
+# To change the background color, the following methods are available:
+# ```crystal
+# "foo".colorize.back(:green)
+# "foo".colorize.on(:green)
+# "foo".colorize.on_green
+# ```
+#
+# It's also possible to change the text decoration:
+# ```crystal
+# "foo".colorize.mode(:underline)
+# "foo".colorize.underline
+# ```
+# 
+# The `colorize` method returns a `Colorize::Object` instance, 
+# which allows chaining methods together:
+# ```crystal
+# "foo".colorize.fore(:yellow).back(:blue).mode(:underline)
+# ```
+#
+# `String` interpolation with `to_s` method:
+# ```crystal
+# "#{"foo".colorize(:red).to_s} bar"
+# #=> "foo bar"
+# #=> "foo" is red
+# ```
```

Comment:
Agree, I also find that `"#{"foo".colorize(:red)} bar"` works too, which makes the `to_s` method not quite important here. I'll take it out.